### PR TITLE
DOC: Improve describe() documentation

### LINKED ISSUE #40972: DOC: Improve describe() doc in getting started
In the "getting started tutorial" here: https://pandas.pydata.org/docs/getting_started/intro_tutorials/01_table_oriented.html

The following sentence is incorrect
Many pandas operations return a DataFrame or a Series. The describe() method is an example of a pandas operation returning a pandas **Series**.

It should say "The describe() method is an example of a pandas operation returning a pandas **DataFrame**"

Steps to reproduce
>>> import pandas as pd
>>> df = pd.DataFrame({
... "Name" : ["Ally", "Bob", "Charly"],
... "Age" : [15,25,36],
... "Cash": [1000,5000, 20000]});
>>> d = df.describe()
>>> type(d)
<class 'pandas.core.frame.DataFrame'>

### PR DESCRIPTION
**Pull request by bharatr21**
- [x] closes #40972 
- [ ] tests added / passed
- [x] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing.html#code-standards) for how to run them
- [ ] whatsnew entry

It feels more verbose, but necessary?

### CODE REVIEW

**Comment by Bharat123rox** [comment]
Test failure seems unrelated

File: doc/source/getting_started/intro_tutorials/01_table_oriented.rst [hunk 1]
```diff
@@ -176,7 +176,7 @@ these are by default not taken into account by the :func:`~DataFrame.describe` m
 
 Many pandas operations return a ``DataFrame`` or a ``Series``. The
 :func:`~DataFrame.describe` method is an example of a pandas operation returning a
-pandas ``Series``.
+pandas ``Series`` (one numerical column) or a pandas ``DataFrame`` (many numerical columns).
```
Comment (phofl) [inline doc/source/getting_started/intro_tutorials/01_table_oriented.rst:179]: Do we need the column reference? Meaning is this done elsewhere in the getting started docs?

**Comment by phofl** [comment]
Yeah this test is flaky

**Comment by mzeitlin11** [comment]
Thanks for looking at this @Bharat123rox. This patch doesn't look to be addressing the issue - the line before already says that different pandas operations might return a Series or DataFrame. 

The issue here is that the next line mentions that this is an example of a Series being returned, but a DataFrame is actually being returned from this describe example

**Comment by Bharat123rox** [comment]
> The issue here is that the next line mentions that this is an example of a Series being returned, but a DataFrame is actually being returned from this describe example

So should we just stick to this example and change it to `DataFrame` because the given `describe()` operation can still return both right? So what should be done then?

**Comment by phofl** [comment]
I would agree with just mentioning DataFrame and Series, e.g. removing the column reference. Do you agree @mzeitlin11?

**Comment by Bharat123rox** [comment]
> I would agree with just mentioning DataFrame and Series, e.g. removing the column reference. Do you agree @mzeitlin11?

Updated, please re-review

**Review by phofl — APPROVED** [review]

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @Bharat123rox

**Pull request merged**